.github/workflows/build-apps.yml

```diff
@@ -115,23 +115,65 @@ jobs:
         uses: actions/checkout@v4
         with:
           submodules: true
+
+      - name: install dependencies
+        run: brew install create-dmg
+
+      - name: Install Apple Certificate
+        uses: apple-actions/import-codesign-certs@v3
+        with:
+          p12-file-base64: ${{ secrets.APPLE_CERTIFICATE_DEVELOPER_ID_APPLICATION_P12 }}
+          p12-password: ${{ secrets.APPLE_CERTIFICATE_PASSWORD }}
+
+      - name: Install the provisioning profile
+        run: |
+          mkdir -p ~/Library/Developer/Xcode/Provisioning\ Profiles
+          echo "${{ secrets.APPLE_MACOS_PROVISION_PROFILE }}" | base64 --decode > ~/Library/Developer/Xcode/Provisioning\ Profiles/Github_Actions.provisionprofile
+
       - name: Set up vcpkg
         uses: lukka/run-vcpkg@v11
         with:
           vcpkgDirectory: ${{ github.workspace }}/app/vcpkg
+
       - name: Set up Flutter
         uses: subosito/flutter-action@v2
         with:
           channel: stable
           flutter-version-file: app/pubspec.yaml
+
       - name: Clean cache
-        run: flutter pub cache clean
-      - run: flutter build macos --dart-define-from-file=${ENV_FILE}
+        run: flutter pub cache clean -f
+
+      - name: Build
+        run: flutter build macos --dart-define-from-file=${ENV_FILE}
+
+      - name: Sign
+        run: ./scripts/macos/sign.sh "${{ secrets.APPLE_DEVELOPER_ID_APPLICATION }}"
+
+      - name: Notarize & Staple
+        run: |
+          echo "${{ secrets.APPLE_KEY_P8 }}" | base64 --decode > ~/apple-key.p8
```

Comment:
Why not inline that in the script?

Reply:
Because this line is specific to this ci environment, while the scripts could be reused in other environments